lib/src/Base/IterativeStat/IterativeThresholdExceedance.cxx

```diff
@@ -0,0 +1,128 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief IterativeThresholdExceedance
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/IterativeThresholdExceedance.hxx"
+#include "openturns/IterativeAlgorithm.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Point.hxx"
+#include "openturns/IterativeThresholdExceedance.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(IterativeThresholdExceedance)
+
+static const Factory<IterativeThresholdExceedance> Factory_IterativeThresholdExceedance;
+
+/** Default constructor */
+IterativeThresholdExceedance::IterativeThresholdExceedance(const UnsignedInteger dimension, const Scalar thresholdValue)
+  : IterativeAlgorithmImplementation(dimension)
+  , thresholdValue_(thresholdValue)
+  , data_(dimension_, 0.0)
+{
+  // Nothing to do
+}
+
+/** Virtual constructor */
+IterativeThresholdExceedance * IterativeThresholdExceedance::clone() const
+{
+  return new IterativeThresholdExceedance(*this);
+}
+
+/** String converter */
+String IterativeThresholdExceedance::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << IterativeThresholdExceedance::GetClassName()
+      << " iteration=" << iteration_
+      << " dimension=" << dimension_
+      << " threshold value=" << thresholdValue_
+      << " values=" << data_.__repr__();
+  return oss;
+}
+
+/** String converter */
+String IterativeThresholdExceedance::__str__(const String & offset) const
+{
+  return data_.__str__(offset);
+}
+
+/** Get the threshold value used */
+Scalar IterativeThresholdExceedance::getThresholdValue() const
+{
+  return thresholdValue_;
+}
+
+/** Get the values exceeding the specified threshold */
+Point IterativeThresholdExceedance::getThresholdExceedance() const
+{
+  return data_;
+}
+
+/** Increment the internal data with a Point */
+void IterativeThresholdExceedance::increment(const Point & newData)
+{
+  if (newData.getSize() != dimension_) throw InvalidArgumentException(HERE) << "Error: the given Point is not compatible with the dimension of the iterative threshold exceedance.";
+  iteration_ += 1;
+  for (UnsignedInteger i = 0; i < dimension_; ++i)
+  {
+    if (newData[i] > thresholdValue_)
+    {
+      data_[i] += 1;
```

Comment:
Either `++data_[i];`  or better `data_[i] += 1.0;` to insist on the floating point nature of data_[i].